specification/azuredatatransfer/resource-manager/Microsoft.AzureDataTransfer/preview/2023-10-11-preview/azuredatatransfer.json

```diff
@@ -0,0 +1,2303 @@
+          }
+        },
+        "operationId": "Pipelines_ListBySubscription",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Listed all pipelines under the subscription.",
+            "schema": {
+              "$ref": "#/definitions/pipelinesListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.AzureDataTransfer/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "operationId": "Operations_List",
+        "x-ms-examples": {
+          "List operations": {
+            "$ref": "./examples/listOperations.json"
+          }
+        },
+        "description": "Lists all of the available Microsoft.AzureDataTransfer REST API operations.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Listed all of the available operations.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/OperationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "actionRequest": {
+      "type": "object",
+      "description": "Payload to perform custom data transfer actions",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the action"
+        },
+        "payload": {
+          "type": "string",
+          "description": "Request messaging describing the action"
+        }
+      }
+    },
+    "actionResult": {
+      "type": "object",
+      "description": "An action result",
+      "properties": {
+        "result": {
+          "type": "string",
+          "description": "Response message describing the action"
+        }
```

Comment:
Are you somehow encoding a HTTP response in a string? That seems like a potential worry.

Reply:
It is a message string describing the result. I have added a longer description and added a `code` property to the result as well to make the purpose of the message clearer.

Reply:
Can you please further explain how anyone using this API knows the valid inputs and outputs to use for 'perform request'?

Reply:
We have updated our API specs to separate out the general action endpoint into more specific provider-level actions.